test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -240,107 +155,133 @@ func schedulePods(config *testConfig) int32 {
+					{
+						MatchExpressions: nodeSelectorRequirements,
 					},
 				},
 			},
-		}
-		podList = append(podList, pod)
+		},
+	}
+}
+
+// generateNodes generates nodes to be used for scheduling.
+func (inputConfig *schedulerPerfConfig) generateNodes(config *testConfig) {
+	for i := 0; i < inputConfig.NodeCount; i++ {
+		config.schedulerSupportFunctions.GetClient().Core().Nodes().Create(config.mutatedNodeTemplate)
+
+	}
+	for i := 0; i < config.numNodes-inputConfig.NodeCount; i++ {
+		config.schedulerSupportFunctions.GetClient().Core().Nodes().Create(getBaseNodeTemplate())
+
 	}
-	return podList
+}
+
+// generatePods generates pods to be used for scheduling.
+func (inputConfig *schedulerPerfConfig) generatePods(config *testConfig) {
+	testutils.CreatePod(config.schedulerSupportFunctions.GetClient(), "sample", inputConfig.PodCount, config.mutatedPodTemplate)
+	testutils.CreatePod(config.schedulerSupportFunctions.GetClient(), "sample", config.numPods-inputConfig.PodCount, getBasePodTemplate())
 }
 
 // generatePodAndNodeTopology is the wrapper function for modifying both pods and node objects.
-func (inputConfig *schedulerPerfConfig) generatePodAndNodeTopology(config *testConfig) {
+func (inputConfig *schedulerPerfConfig) generatePodAndNodeTopology(config *testConfig) error {
+	if config.numNodes < inputConfig.NodeCount || config.numPods < inputConfig.PodCount {
+		fmt.Errorf("NodeCount cannot be greater than numNodes")
+	}
 	nodeAffinity := inputConfig.NodeAffinity
-	podCreatorConfig := testutils.NewTestPodCreatorConfig()
-	var nodeStrategies []testutils.CountToStrategy
-	var pod *v1.Pod
-	var podList []*v1.Pod
+	// Node template that needs to be mutated.
+	mutatedNodeTemplate := getBaseNodeTemplate()
+	// Pod template that needs to be mutated.
+	mutatedPodTemplate := getBasePodTemplate()
 	if nodeAffinity != nil {
-		// Mutate Node
-		nodeStrategies = nodeAffinity.mutateNodeSpec(config.numNodes)
-		// Mutate Pod TODO: Make this to return to podSpec.
-		podList = nodeAffinity.mutatePodSpec(config.numPods, pod)
-		numGroups := nodeAffinity.numGroups
-		for _, pod := range podList {
-			podCreatorConfig.AddStrategy("sched-perf-node-affinity", config.numPods/numGroups,
-				testutils.NewCustomCreatePodStrategy(pod),
-			)
-		}
-		config.nodePreparer = framework.NewIntegrationTestNodePreparer(
-			config.schedulerSupportFunctions.GetClient(),
-			nodeStrategies, "scheduler-perf-")
-		config.podCreator = testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podCreatorConfig)
-		// TODO: other predicates/priorities will be processed in subsequent if statements.
-	} else {
-		// Default configuration.
-		nodePreparer := framework.NewIntegrationTestNodePreparer(
-			config.schedulerSupportFunctions.GetClient(),
-			[]testutils.CountToStrategy{{Count: config.numNodes, Strategy: &testutils.TrivialNodePrepareStrategy{}}},
-			"scheduler-perf-",
-		)
+		// Mutate Node.
+		nodeAffinity.mutateNode(mutatedNodeTemplate)
```

Comment:
lets delete `// Mutate Node.` and `// Mutate Pod.` comments , this code is self explanatory..